### PR TITLE
User ThreadPoolManager to create threadPool and add some prothemus metric about pool

### PR DESCRIPTION
**Pull request by caiconghui**
This PR to to resolve the problem that the usage of CachedPool or FixedThreadPool which not limit thread num or blocked task num  may cause jvm process crashed due to too much thread or oom, we must construct threadpool explicitly, and know the runtime state about threadpool.

### CODE REVIEW

**Comment by caiconghui** [comment]
Ref 3387

File: fe/src/main/java/org/apache/doris/common/ThreadPoolManager.java [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.RejectedExecutionHandler;
+import java.util.concurrent.SynchronousQueue;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
+
+public class ThreadPoolManager {
```
Comment (morningman) [inline fe/src/main/java/org/apache/doris/common/ThreadPoolManager.java:38]: Could add a comment to explain the usage of this class?

File: fe/src/main/java/org/apache/doris/mysql/nio/NMysqlServer.java [hunk 2]
```diff
@@ -47,7 +46,7 @@
     private AcceptingChannel<StreamConnection> server;
 
     // default task service.
-    private ExecutorService taskService = Executors.newCachedThreadPool((new ThreadFactoryBuilder().setDaemon(false).setNameFormat("doris-mysql-nio TASK").build()));
+    private ExecutorService taskService = ThreadPoolManager.newDaemonCacheThreadPool(Config.max_mysql_service_task_threads_num, "doris-mysql-nio-pool");
```
[the remark this answers is not in the document]
Comment (caiconghui) [reply fe/src/main/java/org/apache/doris/mysql/nio/NMysqlServer.java:49]: It doesn't matter, for thread which submit task is not a daemon thread. for the whole fe jvm process, only then there are no non-daemon threads will cause jvm process exist

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**